### PR TITLE
fix(table): sortable, titleTooltip, clear selections on page change

### PR DESCRIPTION
**Pull request by skjindal93**

### CODE REVIEW

File: projects/components/src/table/table.component.ts [hunk 1]
```diff
@@ -603,6 +616,10 @@ export class TableComponent
         }
       : undefined;
   }
+
+  private isColumnSortable(columnConfig: TableColumnConfig): boolean {
+    return columnConfig.sortable === undefined || columnConfig.sortable;
```
Comment (arjunlalb) [inline projects/components/src/table/table.component.ts:621]: shouldn't this be `value !== undefined && sortable` ? also, is `sortable` a valid property? shouldn't it be `sort` ?

Comment (skjindal93) [reply projects/components/src/table/table.component.ts:621]: Actually, I wanted it to have sort by default, even if someone does not add sortable property in the column config

File: projects/components/src/table/table.component.ts [hunk 2]
```diff
@@ -345,10 +352,12 @@ export class TableComponent
   }
 
   public onHeaderCellClick(columnConfig: TableColumnConfig): void {
-    this.updateSort({
-      column: columnConfig,
-      direction: this.getNextSortDirection(columnConfig.sort)
-    });
+    if (this.isColumnSortable(columnConfig)) {
```
Comment (aaron-steinfeld) [inline projects/components/src/table/table.component.ts:355]: Is sortable not used anywhere any more? must be a regression. We should also change the styling for not sortable so it doesn't have a hover pointer.

File: projects/components/src/table/table.component.ts [hunk 3]
```diff
@@ -562,6 +571,10 @@ export class TableComponent
         [TableComponent.PAGE_SIZE_URL_PARAM]: pageEvent.pageSize
       });
     }
+
+    this.selections = [];
+    this.selectionsChange.emit(this.selections);
```
Comment (aaron-steinfeld) [inline projects/components/src/table/table.component.ts:576]: This should only emit if there were previously selections - that is, it changed.

Comment (skjindal93) [reply projects/components/src/table/table.component.ts:576]: Just for optimization, right?

Comment (aaron-steinfeld) [reply projects/components/src/table/table.component.ts:576]: For correctness - this is a change emitter, it should only emit on change.

File: projects/components/src/table/table.component.ts (hunk 2 above)
Comment (skjindal93) [reply projects/components/src/table/table.component.ts:355]: Right. The sortable was behaving as a no-op (_as in it was always sorting, irrespective of the value_)

File: projects/components/src/table/data/table-cdk-column-util.ts [hunk 4]
```diff
@@ -17,5 +17,11 @@ export namespace TableCdkColumnUtil {
   export const unsortOtherColumns = (sortedColumn: TableColumnConfig, otherColumns: TableColumnConfig[]): void =>
     otherColumns
       .filter(column => column.field !== sortedColumn.field)
-      .forEach(filteredColumn => (filteredColumn.sort = undefined));
+      .forEach(filteredColumn => {
+        filteredColumn.sort = undefined;
+        filteredColumn.sortable = false;
```
Comment (anandtiwary) [inline projects/components/src/table/data/table-cdk-column-util.ts:22]: Should we set `sortable = false`? Wouldn't this take out the sort feature of an existing column?

Comment (skjindal93) [reply projects/components/src/table/data/table-cdk-column-util.ts:22]: Shit. Yes. Nice catch!

**Pull request merged**